cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -267,15 +269,26 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 	unregisteredNodes := a.clusterStateRegistry.GetUnregisteredNodes()
 	if len(unregisteredNodes) > 0 {
 		klog.V(1).Infof("%d unregistered nodes present", len(unregisteredNodes))
-		removedAny, err := removeOldUnregisteredNodes(unregisteredNodes, autoscalingContext,
-			a.clusterStateRegistry, currentTime, autoscalingContext.LogRecorder)
-		// There was a problem with removing unregistered nodes. Retry in the next loop.
-		if err != nil {
-			klog.Warningf("Failed to remove unregistered nodes: %v", err)
-		}
-		if removedAny {
-			klog.V(0).Infof("Some unregistered nodes were removed, skipping iteration")
-			return nil
+
+		// Add protection for remove unregister nodes
+		if a.lastRemoveUnregister.Add(a.RemoveUnregisterDelay).After(currentTime) {
+			for _, unregisteredNode := range unregisteredNodes {
+				klog.V(4).Infof("Skipping remove unregister node: %v, lastRemoveUnregister=%s",
```

Comment:
It feels like this should have more explicit information about why the nodes were skipped. The time of the last removal without any further info wouldn't be very clear on first read.

Reply:
> It feels like this should have more explicit information about why the nodes were skipped. The time of the last removal without any further info wouldn't be very clear on first read.

Thanks  a lot @gjtempleton 
Excellent.I accept your ideas, I have made some changes .  

About the why the nodes were skipped, as my opinion it is similar with ScaleDownDelayAfterDelete . More info is not necessary if we know what is UnregisteredNodeRemovalDelay. 

 We see 'Skipping remove unregister node..', UnregisteredNodeRemovalDelay time not reached

1  UnregisteredNodeRemovalDelay could be optional to reduce the probability of wrong removal
when we have so many k8s clusters

2 If we have chance to trigger some custom behavior  during in node removal delay time,  eg: monitor alarm (if necessary), that also be nice

I add more comments in new commit